### PR TITLE
updated hypershift option to include external-dns option

### PR DESCRIPTION
**Pull request by mukrishn**
### Description
By default HostedClusters control plane traffic will get routed via management cluster router pods and that gets over utilized when running more clusters, on hosting 80 HC the usage increased up to 12-14 times actual requested pod resource and that might affect management cluster router performance as well.
This will be reduced if we have dedicated router pods for each HC, by adding external-dns-domain option will provision HA router pods within HCP namespace per HC. Hypershift when GA will do this by default when creating cluster using rosa cli.
Adding this in to our automation to have all HCP pods for our management testing, hc_external_dns can be set only if hc is >= 4.12.

### CODE REVIEW

File: workloads/hyper-scale/README.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ The run.sh script can be tweaked with the following environment variables
 | **ROSA_TOKEN** | ROSA token for access, only staging account token is valid currently | **REQUIRED** |
 | **PULL_SECRET**   | Cloud pull secret for Openshift installation  | **REQUIRED** |
 | **NUMBER_OF_HOSTED_CLUSTER**         | Integer: number of hosted cluster to be deployed on given management cluster | `2` |
-| **COMPUTE_WORKERS_NUMBER**         | Integer: number of workers nodes to be created on each hosted cluster | `24` |
+| **COMPUTE_WORKERS_NUMBER**         | Integer: number of workers nodes to be created on each hosted cluster. This value will be consider as per zone number when `HC_MULTI_AZ` is set, so this default value creates 3 nodes totally | `1` |
```
Comment (dry923) [inline workloads/hyper-scale/README.md:34]: Needs rewording. I'd start at "Integer: number of worker nodes PER Availability Zone"

File: workloads/hyper-scale/common.sh [hunk 2]
```diff
@@ -40,7 +40,21 @@ prep(){
 
 setup(){
     export MGMT_CLUSTER_NAME=$(oc get infrastructure cluster -o jsonpath='{.status.infrastructureName}'| cut -c 1-13)
-    export BASEDOMAIN=$(oc get dns cluster -o jsonpath='{.spec.baseDomain}')
+    export MGMT_BASEDOMAIN=$(oc get dns cluster -o jsonpath='{.spec.baseDomain}')
+    export MGMT_AWS_HZ_ID=$(aws route53 list-hosted-zones | jq -r '.HostedZones[] | select(.Name=="'${MGMT_BASEDOMAIN}'.")' | jq -r '.Id')
+    if [[ $HC_EXTERNAL_DNS != "false" ]]; then
```
Comment (dry923) [inline workloads/hyper-scale/common.sh:45]: Nit but can we just say if its true lol

Comment (dry923) [reply workloads/hyper-scale/common.sh:45]: Also, we should check the openshift version here as well to make sure we arent attempting w/ a < 4.12 image

File: workloads/hyper-scale/common.sh [hunk 3]
```diff
@@ -74,10 +88,23 @@ install(){
...
     aws s3api wait bucket-exists --bucket $MGMT_CLUSTER_NAME-aws-rhperfscale-org 
     HO_IMAGE_ARG=""
+    HCP_P_MONITOR=""
+    EXT_DNS_ARG=""    
     if [[ $HYPERSHIFT_OPERATOR_IMAGE != "" ]]; then
             HO_IMAGE_ARG="--hypershift-image $HYPERSHIFT_OPERATOR_IMAGE"
-    fi    
-    hypershift install $HO_IMAGE_ARG --oidc-storage-provider-s3-bucket-name $MGMT_CLUSTER_NAME-aws-rhperfscale-org --oidc-storage-provider-s3-credentials aws_credentials --oidc-storage-provider-s3-region $AWS_REGION  --platform-monitoring All --metrics-set All
+    fi
+    if [[ $HCP_PLATFORM_MONITORING != "false" ]]; then
```
Comment (dry923) [inline workloads/hyper-scale/common.sh:96]: Nit, again how about if its true :-D

File: workloads/hyper-scale/common.sh [hunk 4]
```diff
@@ -40,7 +40,30 @@ prep(){
 
 setup(){
     export MGMT_CLUSTER_NAME=$(oc get infrastructure cluster -o jsonpath='{.status.infrastructureName}'| cut -c 1-13)
-    export BASEDOMAIN=$(oc get dns cluster -o jsonpath='{.spec.baseDomain}')
+    export MGMT_BASEDOMAIN=$(oc get dns cluster -o jsonpath='{.spec.baseDomain}')
+    export MGMT_AWS_HZ_ID=$(aws route53 list-hosted-zones | jq -r '.HostedZones[] | select(.Name=="'${MGMT_BASEDOMAIN}'.")' | jq -r '.Id')
```
Comment (krishvoor) [inline workloads/hyper-scale/common.sh:44]: My `export AWS_DEFAULT_OUTPUT="text"` as default,  and ran into issues.
let's explicitly specify the `--output json`  something like this,
```
aws route53 list-hosted-zones --output json | jq -r '.HostedZones[] | select(.Name=="'${MGMT_BASEDOMAIN}'.")' | jq -r '.Id
```

File: workloads/hyper-scale/common.sh [hunk 5]
```diff
@@ -40,7 +40,30 @@ prep(){
...
+    if [[ $HC_EXTERNAL_DNS == "true" ]]; then
+        # check given Hypershift operator version is >= 4.12
+        OP_REL_CHECK=$(echo "$(echo $HYPERSHIFT_OPERATOR_IMAGE | awk -F: '{print$2}' | cut -c 1-4) >= 4.12" |bc -l)
+        # check given HostedCluster release is >= 4.12
+        REL_CHECK=$(echo "$(echo $RELEASE_IMAGE | awk -F: '{print$2}' | cut -c 1-4) >= 4.12" |bc -l)
+        if [[ $REL_CHECK == 1 ]] || [[ "$RELEASE_IMAGE" == "" && $(echo $HYPERSHIFT_OPERATOR_IMAGE | awk -F: '{print$2}' | cut -c 1-4) == "late" || $OP_REL_CHECK == 1 ]]; then
+            echo "Create external DNS for this iteration.."
+            export BASEDOMAIN=hyp.${MGMT_BASEDOMAIN}
+            AWS_HZ=$(aws route53 list-hosted-zones | jq -r '.HostedZones[] | select(.Name=="'${BASEDOMAIN}'.")')
```
Comment (krishvoor) [inline workloads/hyper-scale/common.sh:53]: Here too

File: workloads/hyper-scale/common.sh [hunk 6]
```diff
@@ -40,7 +40,30 @@ prep(){
...
+            if [[ ${AWS_HZ} == "" ]]; then
+                AWS_HZ_ID=$(aws route53 create-hosted-zone --name $BASEDOMAIN --caller-reference ${MGMT_CLUSTER_NAME}-$(echo $(uuidgen) | cut -c 1-5) | jq -r '.HostedZone.Id')
+                DS_VALUE=$(aws route53 list-resource-record-sets --hosted-zone-id $AWS_HZ_ID  | jq -r '.ResourceRecordSets[] | select(.Name=="'"$BASEDOMAIN"'.") | select(.Type=="NS")' | jq -c '.ResourceRecords')
```
Comment (krishvoor) [inline workloads/hyper-scale/common.sh:56]: Here too

**Review by krishvoor — COMMENTED** [review]
Overall lgtm.

**Comment by mukrishn** [comment]
@dry923, @krishvoor  PTAL

**Review by dry923 — APPROVED** [review]
LGTM!

File: workloads/hyper-scale/common.sh [hunk 7]
```diff
@@ -56,7 +79,7 @@ setup(){
 
 pre_flight_checks(){
     echo "Pre flight checks started"
-    export MULTI_AZ=$(rosa describe cluster -c $MGMT_CLUSTER_NAME -o json | jq -r [.multi_az] | jq -r .[])
+    export MULTI_AZ=$(ocm describe cluster $MGMT_CLUSTER_NAME --json | jq -r [.multi_az] | jq -r .[])
```
Comment (krishvoor) [inline workloads/hyper-scale/common.sh:82]: I observed this on non-rosa cluster, good catch..

**Review by krishvoor — APPROVED** [review]
lgtm

**Pull request merged**